lib/puppet/type/file.rb

```diff
@@ -851,10 +851,14 @@ def write(property = nil)
     remove_existing(:file)
 
     mode = self.should(:mode) # might be nil
+    owner = self.should(:owner)
+    group = self.should(:group)
     mode_int = mode ? symbolic_mode_to_int(mode, Puppet::Util::DEFAULT_POSIX_MODE) : nil
+    uid = owner ? self.provider.name2uid(owner) : nil
+    gid = group ? self.provider.name2gid(owner) : nil
 
     if write_temporary_file?
-      Puppet::Util.replace_file(self[:path], mode_int) do |file|
+      Puppet::Util.replace_file(self[:path], mode_int, uid, gid) do |file|
```

Comment:
rather than passing these in to `replace_file`, we have access to them in the block being passed, which is really the only place where we need them.

Below, in the if block for `:validate_cmd`, we can enforce state.

So something like 
```
         if self[:validate_cmd]
           file.chmod(mode_int)
           file.chown(uid, gid)
           output = Puppet::Util::Execution.execute(self[:validate_cmd].gsub(self[:validate_replacement], file.path), :failonfail => true, :combine => true)
           output.split(/\n/).each { |line|
             self.debug(line)
           }
         end
```

this means we don't have to modify the `replace_file` method.

Are there other cases where you'd want to enforce perms on the tempfile beyond the `validate_cmd` scenario?

cc/ @jtappa

Reply:
@Iristyle can you or someone from the windows team advise on how best to manage this for windows? Is this something we can address on windows, or should we not attempt to manage the perms/ownership of the temp file if we're on windows?